Semver/Ranges/RangeParseException.cs

```diff
@@ -0,0 +1,18 @@
+using System;
+
+namespace Semver.Ranges
+{
+    /// <summary>
+    /// The exception that is thrown when a range's syntax could not be parsed.
+    /// </summary>
+    public class RangeParseException : Exception
```

Comment:
A custom exception shouldn't be needed. `FormatException` is the standard exception for parsing errors.